docs/DeveloperGuide.md

```diff
@@ -1,12 +1,220 @@
 
-## Design & implementation
+Note: The `Carpark` class contain many getters, setters and annotations to be used with the `jackson` module. 
+See [`FileLoader`](#FileLoader) for more information.
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+#### API Component
+![API Class Diagram](images/ApiClassDiagram.png)
+
+The model component consists of a `Api` class that supports API call to LTA DataMall Services, under the
+package `seedu.api`.
+
+This component:
+* Focuses on communicating with an external Api by using HTTP requests. 
+* JSON response will be stored in the local JSON file as long as the response is `200 OK` and not empty.
+* Only reads from the secret.txt file once and stores the API as a variable (will only read from the file
+if requested again).
+
+How the API component of data fetching works
+1. The initialisation of the class will make an instance of the `Storage`, `HttpClient` and `Ui` class.
+2. If the program just started, the Parking class will call the `loadApiKey` method.
+   - If no file called `secret.txt` is found, file will be created.
+   - If the file is empty, default key will be loaded to the Api instance (but not written to the local file).
+3. To get a certain data set from the Api Service,
+   - Call `asyncExecuteRequest` method to construct the HTTP request packet header and sends the request
+     asynchronously.
+   - Call `fetchData` method to get response immediately from the Api.
+      - This method will also validate the response from the Api by identifying the response code.
+      - Unless its 401 Unauthorised Access (due to invalid Api key), the method will try to fetch data
+        at most 5 times.
+
+Since each API call only returns 500 data, we need to make multiple api calls. Thus, there is another function 
+that handles the API calls all at once. The process is still similar.
+1. Same step 1 and 2 above.
+2. Call `syncFetchData` method, which calls both the `asyncExecuteRequest` and `fetchData` five times (LTA only has 2312 
+Parking lot data)
+   - This method makes all 5 request asynchronously and receives in sequence.
+   - The method will concatenate the data together for data processing by other parts of the program.
+   - The processed data will then be stored in a local data file.
+   
+The following sequence diagram shows how data is fetched using the `update` command.
+![Update Data API](images/UpdateApiSequenceDiagram.png)
+
+
+The API component is also able to:
+- Loads in key from a local file storage (in txt format).
+- Authenticate user API key. If no user key inputted, default key will be loaded.
+- Get API authentication status.
+
+The following sequence diagram shows how the API key is loaded.
+![Api Loading Sequnce Diagram](images/LoadApiSequenceDiagram.png)
+
+
+#### Storage Component
+
+#### Implementation
+![Sequence Diagram](images/LoadFileSequenceDiagram.png)
```

Comment:
This sequence diagram seems overly complicated, do try to re-visit whether it is necessary to depict every aspect of the sequence diagram. If it is requried, you can potentially consider splitting it up into multiple diagrams, or using ref frames.